### PR TITLE
PBS Adapter: Add PBS_ANALYTICS Event

### PR DESCRIPTION
**Pull request by spotxslagle**
## Type of change

- [ x ] Feature

## Description of change

Add PBS_ANALYTICS event and pass ATAG data through it

## Other information

Supercedes https://github.com/prebid/Prebid.js/pull/12015

### CODE REVIEW

**Comment by spotxslagle** [comment]
fixes https://github.com/prebid/Prebid.js/issues/11391

**Comment by JulieLorin** [comment]
I was reading the linked issue and I see `We should emit a new event for the server side analytics object being available, and only emit the event when the library is configured by the publisher to do so `
Not sure what `when the library is configured by the publisher to do so` exactly means but shouldn't that means there should be a configuration value ? (Or is that only for server side ?)

Otherwise, the code looks good

File: modules/prebidServerBidAdapter/index.js
Comment (osazos) [inline modules/prebidServerBidAdapter/index.js:617]: The naming sounds confusing, like a function that returns a bool that allows to emit the "analytics" event. 
Can we try to find something better?
Proposition:  `allowedPbsEvents`, `getPbsEventsEmitStatus`, `emitPbsEvents`, something like that.

File: modules/prebidServerBidAdapter/index.js
Comment (osazos) [inline modules/prebidServerBidAdapter/index.js:492]: consistency:

```suggestion
              seatnonbid: response?.ext?.seatnonbid,
              atag: response?.ext?.prebid?.analytics?.tags,
```

If the ternary condition is related to the "OR" condition above, there is perhaps an issue because you can still have access to the values in the `data.response` field.

**Comment by patmmccann** [comment]
@spotxslagle should we also change generic analytics adapter to listen for this as in scope here?

**Review by robertrmartinez — APPROVED** [review]
LGTM

Regarding hiding behind config flag I do not think it is necessary since the publisher would be working with a PBS which they may expect or not to set these EXT objects. But if we feel more comfortable hiding this logic I guess we can.

These events only fire if the PBS response contains the EXTs.

**Comment by patmmccann** [comment]
>Regarding hiding behind config flag I do not think it is necessary since the publisher would be working with a PBS which they may expect or not to set these EXT objects.

That's a great point, seems like a good reason to default to true?

**Comment by robertrmartinez** [comment]
yeah - imo that as well is overkill.

@osazos could you give your re-review after @spotxslagle updated the function names etc? Thanks!

**Review by osazos — APPROVED** [review]
LGTM

@spotxslagle, seen after approved: If you can fix the name of the function ~~`getNonbidData`~~ -> `getNonBidData` in a consistency matter only. _(btw this is cosmetic)_

**Comment by osazos** [comment]
@spotxslagle can you fix the function name related to my last comment?
@ChrisHuie do you want to add your review?

**Comment by spotxslagle** [comment]
> @spotxslagle can you fix the function name related to my last comment? @ChrisHuie do you want to add your review?

Done!

**Pull request merged**
